### PR TITLE
JAMES-2952 SieveExecutor::handleFailure should position SIEVE_NOTIFIC…

### PR DESCRIPTION
**Pull request by chibenwa**
…ATION attribute

This prevents double recomputation...

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/delivery/SieveExecutor.java [hunk 1]
```diff
@@ -154,8 +154,17 @@ private void sieveMessageEvaluate(MailAddress recipient, Mail aMail, ResourceLoc
         }
     }
 
-    protected void handleFailure(MailAddress recipient, Mail aMail, Exception ex) throws MessagingException, IOException {
+    void handleFailure(MailAddress recipient, Mail aMail, Exception ex) throws MessagingException, IOException {
```
Comment (rouazana) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/delivery/SieveExecutor.java]: @VisibleForTesting?

**Comment by rouazana** [comment]
an integration test would have been very welcome also

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/delivery/SieveExecutor.java [hunk 2]
```diff
@@ -154,8 +155,18 @@ private void sieveMessageEvaluate(MailAddress recipient, Mail aMail, ResourceLoc
         }
     }
 
-    protected void handleFailure(MailAddress recipient, Mail aMail, Exception ex) throws MessagingException, IOException {
+    @VisibleForTesting
+    void handleFailure(MailAddress recipient, Mail aMail, Exception ex) throws MessagingException, IOException {
         aMail.setAttribute(new Attribute(SIEVE_NOTIFICATION, AttributeValue.of(true)));
```
Comment (mbaechler) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/delivery/SieveExecutor.java]: why do we set the attribute `SIEVE_NOTIFICATION` the mail for which we have the failure?

**Comment by chibenwa** [comment]
The pb is that we wokld fix the cause of the failure straight away, what is brittle.

Unless we add a fail instruction in ksieve but it is more work...

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/delivery/SieveExecutor.java (hunk 2 above)
Comment (chibenwa) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/delivery/SieveExecutor.java]: I should remove this line sorry

**Comment by Arsnael** [comment]
```
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb] [ERROR] Failures: 
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb] [ERROR]   SieveIntegrationTest.doubleVacationShouldNotBeExecutedAndReceiverShouldHaveANotificationAboutSieveError:809 
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb] Expecting:
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb]   <[SentMail{recipients=[receiver@domain.com], sender=receiver@domain.com, attributeNames={AttributeName{name=SieveNotification}=Attribute{name=AttributeName{name=SieveNotification}, value=AttributeValue{value=true, serializer=BooleanSerializer}}, AttributeName{name=org.apache.james.SentByMailet}=Attribute{name=AttributeName{name=org.apache.james.SentByMailet}, value=AttributeValue{value=true, serializer=BooleanSerializer}}}, state=root}]>
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb] to contain exactly (and in same order):
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb]   <[SentMail{recipients=[receiver@domain.com], sender=receiver@domain.com, attributeNames={AttributeName{name=org.apache.james.SentByMailet}=Attribute{name=AttributeName{name=org.apache.james.SentByMailet}, value=AttributeValue{value=true, serializer=BooleanSerializer}}}, state=root}]>
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb] but some elements were not found:
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb]   <[SentMail{recipients=[receiver@domain.com], sender=receiver@domain.com, attributeNames={AttributeName{name=org.apache.james.SentByMailet}=Attribute{name=AttributeName{name=org.apache.james.SentByMailet}, value=AttributeValue{value=true, serializer=BooleanSerializer}}}, state=root}]>
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb] and others were not expected:
[1d9547689c4d69db1814c5aa4a29493f92cfe9eb]   <[SentMail{recipients=[receiver@domain.com], sender=receiver@domain.com, attributeNames={AttributeName{name=SieveNotification}=Attribute{name=AttributeName{name=SieveNotification}, value=AttributeValue{value=true, serializer=BooleanSerializer}}, AttributeName{name=org.apache.james.SentByMailet}=Attribute{name=AttributeName{name=org.apache.james.SentByMailet}, value=AttributeValue{value=true, serializer=BooleanSerializer}}}, state=root}]>
```

**Comment by chibenwa** [comment]
```
RabbitMQEventBusTest$ConcurrentTest.concurrentDispatchKeyShouldDeliverAllEventsToListenersWithSingleEventBus » AlreadyClosed

05:40:07.662 [ERROR] o.a.j.u.c.ConcurrentTestRunner - Error caught during concurrent testing (iteration 4159, threadNumber 4)
java.lang.IllegalStateException: Event Bus is not running
```

test this please

**Comment by chibenwa** [comment]
```
RabbitMQEventBusTest$ConcurrentTest.concurrentDispatchKeyShouldDeliverAllEventsToListenersWithSingleEventBus » AlreadyClose

[88d7e4736e39a75058f168ae217549ada9884623] 05:40:07.877 [ERROR] o.a.j.u.c.ConcurrentTestRunner - Error caught during concurrent testing (iteration 4226, threadNumber 4)
[88d7e4736e39a75058f168ae217549ada9884623] java.lang.IllegalStateException: Event Bus is not running
```

IMO could be that it takes too long..

test this please

**Comment by Arsnael** [comment]
Merged.

**Pull request closed without merging**